src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -509,7 +509,7 @@ private void ScanSyntaxToken(ref TokenInfo info)
 
                 case '!':
                     TextWindow.AdvanceChar();
-                    if (TextWindow.PeekChar() == '=')
+                    if (TextWindow.PeekChar() == '=' && TextWindow.PeekChar(1) != '>')
```

Comment:
>if (TextWindow.PeekChar() == '=' && TextWindow.PeekChar(1) != '>') [](start = 20, length = 66)

It looks like we're treating `!=>` as `! =>` rather than `!= >`. I believe we always choose the longest possible token so we should prefer `!= >`.